Support `HtmlString` in repeater and builder `itemLabel`

bernhardh opened the pull request:
This PR adds an item label to the Builder. Its basically just a copy of the functionality already existing on the repeater and is used in the same way:

```php
Forms\Components\Builder::make("xxx")
    ->itemLabel(fn (array $state): ?string => $state['title'] ?? null)
```

This is very usefull, when sorting long lists, by collapsing:

![image](https://user-images.githubusercontent.com/642292/231768077-4872546b-74ac-4dfe-b118-817237e7dce8.png)

bernhardh committed:
ADD: Builder itemLabel

danharrin commented:
I think `->itemLabel()` should be on the `Block` level instead of the `Builder` level, because then you can have a different label pattern for each block which is good for different data structures.

zepfietje commented:
Are you able to implement the requested changes, @bernhardh? 🙂

bernhardh committed:
Merge branch 'filamentphp:2.x' into feature/add-builder-item-label

zepfietje commented:
Great, @bernhardh, thanks!

bernhardh committed:
CHG: Move itemLabel to block level for Builder

bernhardh committed:
Merge remote-tracking branch 'origin/feature/add-builder-item-label' into feature/add-builder-item-label

bernhardh committed:
chore: fix code style

bernhardh commented:
@zepfietje 
I think I have changed it as suggested. You can now set the itemLabel on `Block`  level instead of globally for the whole `Builder`.

One minor thing I changed compared to the behavior of `Repeater`: You can also use/retour `HtmlString`'s:

```php
->itemLabel(fn (array $state): HtmlString => new HtmlString(
    $state['title'] ?? "<strong class='text-red-500'>TITLE MISSING!!!</strong>"
))
```

danharrin commented on packages/forms/resources/views/components/builder.blade.php:
```diff
@@ -133,8 +138,6 @@
                                         @php
-                                            $block = $item->getParentComponent();
-
                                             $block->labelState($item->getRawState());
```
Wait... I think this feature already exists using the existing `label()` method on the block.

bernhardh replied:
Lol. You are totally right! I was so focused on the Builder level, that I never thought about the block level, because this is how it works on the Repeater. 

So my complete PR is useless. Only missing feature is the HtmlString instead of just plain string. Can/should I close this PR and create a new one to add HtmlString to both - Block label and Repeater itemLabel?

zepfietje replied:
Sure!

danharrin replied:
Or you can use this PR, it doesn't matter

bernhardh commented:
Closed in favor of #6321
